src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/RadioactivityTask.java

```diff
@@ -0,0 +1,161 @@
+    public static void removePlayer(Player p){
+        radioactivityLevel.remove(p.getUniqueId());
+    }
+    private final PotionEffect WITHER = new PotionEffect(PotionEffectType.WITHER,
+            SlimefunPlugin.getCfg().getOrSetDefault("options.radiation-update-interval", 1) * 20 + 20,
+            1);
+    private final PotionEffect WITHER2 = new PotionEffect(PotionEffectType.WITHER,
+            SlimefunPlugin.getCfg().getOrSetDefault("options.radiation-update-interval", 1) * 20 + 20,
+            4);
+    private final PotionEffect BLINDNESS = new PotionEffect(PotionEffectType.BLINDNESS,
+            SlimefunPlugin.getCfg().getOrSetDefault("options.radiation-update-interval", 1) * 20 + 40,
+            0);
+    private final PotionEffect SLOW = new PotionEffect(PotionEffectType.SLOW,
+            SlimefunPlugin.getCfg().getOrSetDefault("options.radiation-update-interval", 1) * 20 + 20,
+            3);
+
+    @Override
+    public void run() {
+        for (Player p: Bukkit.getOnlinePlayers()){
+            if (!p.isValid() || p.isDead()) {
+                continue;
+            }
+
+            PlayerProfile.get(p, profile -> handleRadiation(p, profile));
+        }
+    }
+
+    private void handleRadiation(@Nonnull Player p, @Nonnull PlayerProfile prof){
+        Set<SlimefunItem> radioactiveItems = SlimefunPlugin.getRegistry().getRadioactiveItems();
+        if (p.getGameMode() == GameMode.CREATIVE || p.getGameMode() == GameMode.SPECTATOR) return;
+        int exposureTotal = 0;
+        UUID u = p.getUniqueId();
+        if (!prof.hasFullProtectionAgainst(ProtectionType.RADIATION)){
+            for (ItemStack item : p.getInventory()) {
+                if (item == null || item.getType() == Material.AIR) continue;
+                ItemStack tmpItem = item;
+
+                if (!(item instanceof SlimefunItemStack) && radioactiveItems.size() > 1) {
+                    // Performance optimization to reduce ItemMeta calls
+                    tmpItem = ItemStackWrapper.wrap(tmpItem);
+                }
+                for (SlimefunItem i: radioactiveItems){
+                    if (i.isItem(tmpItem)){
+                        exposureTotal += tmpItem.getAmount() * ((RadioactiveItem) i).getRadioactivity().exposureModifier;
+                    }
+                }
+            }
+        }
+        int exposureLevelBefore = radioactivityLevel.getOrDefault(u, 0);
+        if (exposureTotal > 0){
+            if (exposureLevelBefore == 0){
+                SlimefunPlugin.getLocalization().sendMessage(p, "messages.radiation-exposed");
+            }
+            radioactivityLevel.put(u, Math.min(exposureLevelBefore + exposureTotal, 100));
+        } else if (exposureLevelBefore > 0) {
+            radioactivityLevel.put(u, Math.max(exposureLevelBefore - 1, 0));
+        }
+        int exposureLevelAfter = radioactivityLevel.getOrDefault(u, 0);
+        for (Symptom symptom: Symptom.values()){
```

Comment:
`values()` will make a new array EVERY call. This is intensive and not good. We should instead have a value cache. Check some other enum like Instruction for this

```suggestion
        for (Symptom symptom : Symptom.values()) {
```

Reply:
Sorta guessed so but wasn't sure